src/Api/Helpers/FriendlyExceptionsMiddleware.cs

```diff
@@ -49,6 +50,21 @@ private async Task SafeNext(HttpContext context)
                 type: $"https://docs.passwordless.dev/guide/errors.html#{apiException.ErrorCode}");
             await problem.ExecuteAsync(context);
         }
+        catch (ValidationException validationException)
+        {
+            Tracer.Instance.ActiveScope?.Span.SetException(validationException);
+            _logger.UncaughtException(validationException);
+
+            await Results.ValidationProblem(
+                validationException.Errors
+                    .GroupBy(x => x.PropertyName)
+                    .ToDictionary(x => x.Key, x => x.Select(e => e.ErrorMessage).ToArray()),
+                validationException.Message,
+                validationException.Source,
+                StatusCodes.Status400BadRequest,
+                "One or more validation errors occured"
+            ).ExecuteAsync(context);
+        }

```

Comment:
I'm fine with this generally but only for exceptions thrown outside of the endpoint context, like if we have to do more validation inside a service then we can throw and have them caught here. But if we have the validation problems inside the endpoint we can just return them from there and avoid throwing an exception.

Reply:
Thank you. I saw the other two exceptions and didn't know if this was the pattern we wanted.  I've removed the validation block.